clang/lib/Driver/ToolChains/SYCL.cpp

```diff
@@ -475,17 +475,36 @@ static void addImpliedArgs(const llvm::Triple &Triple,
                            const llvm::opt::ArgList &Args,
                            llvm::opt::ArgStringList &CmdArgs) {
   // Current implied args are for debug information and disabling of
-  // optimizations.
-  // TODO: Add support for other architectures (gen, x86_64) as those are
-  // being defined.
+  // optimizations.  They are passed along to the respective areas as follows:
+  //  FPGA and default device:  -g -cl-opt-disable
+  //  GEN:  -options "-g -O0"
```

Comment:
I think this it true only for Level0, but OpenCL should accept -cl-opt-disable. Right?

I think we should add OpenCL/Level0 to target triple to differentiate between two execution environments (similar to NVCL/CUDA).

Reply:
If not a different triple, maybe an additional option? (can be discussed) What do you think @kbobrovs?  A new report can be created to flesh it out.

Reply:
> A new report can be created to flesh it out.

Okay. I'll merge this PR. Let's address this separately.